Tests/Charts/BarChartTests.swift

```diff
@@ -4,88 +4,341 @@ import FBSnapshotTestCase
 
 class BarChartTests: FBSnapshotTestCase
 {
-    
-    var chart: BarChartView!
-    var dataSet: BarChartDataSet!
-    
     override func setUp()
     {
         super.setUp()
         
         // Set to `true` to re-capture all snapshots
         self.recordMode = false
-        
-        // Sample data
-        let values: [Double] = [8, 104, 81, 93, 52, 44, 97, 101, 75, 28,
-            76, 25, 20, 13, 52, 44, 57, 23, 45, 91,
-            99, 14, 84, 48, 40, 71, 106, 41, 45, 61]
-        
+    }
+    
+    override func tearDown()
+    {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+    
+    //MARK: Prepare
+    func setupCustomValuesDataEntries(values: [Double]) -> [ChartDataEntry]
+    {
         var entries: [ChartDataEntry] = Array()
-        
         for (i, value) in values.enumerated()
         {
             entries.append(BarChartDataEntry(x: Double(i), y: value, icon: UIImage(named: "icon", in: Bundle(for: self.classForCoder), compatibleWith: nil)))
         }
-        
-        dataSet = BarChartDataSet(values: entries, label: "Bar chart unit test data")
+        return entries
+    }
+
+    func setupDefaultValuesDataEntries() -> [ChartDataEntry]
+    {
+        let values: [Double] = [8, 104, -81, 93, 52, -44, 97, 101, -75, 28,
+                                -76, 25, 20, -13, 52, 44, -57, 23, 45, -91,
+                                99, 14, -84, 48, 40, -71, 106, 41, -45, 61]
+        return setupCustomValuesDataEntries(values: values)
+    }
+
+    func setupPositiveValuesDataEntries() -> [ChartDataEntry]
+    {
+        let values: [Double] = [8, 104, 81, 93, 52, 44, 97, 101, 75, 28,
+                                76, 25, 20, 13, 52, 44, 57, 23, 45, 91,
+                                99, 14, 84, 48, 40, 71, 106, 41, 45, 61]
+        return setupCustomValuesDataEntries(values: values)
+    }
+
+    func setupNegativeValuesDataEntries() -> [ChartDataEntry]
+    {
+        let values: [Double] = [-8, -104, -81, -93, -52, -44, -97, -101, -75, -28,
+                                -76, -25, -20, -13, -52, -44, -57, -23, -45, -91,
+                                -99, -14, -84, -48, -40, -71, -106, -41, -45, -61]
+        return setupCustomValuesDataEntries(values: values)
+    }
+
+    func setupZeroValuesDataEntries() -> [ChartDataEntry]
+    {
+        let values: [Double] = [0,0,0,0,0,0,0,0,0,0,
```

Comment:
`let values = [Double](repeating: 0.0, count: 15)`

Reply:
@jjatie I found a weird issue, the old values count is 30, if I use your 15 or even count: 1, the UT still passes, which should fail. Can you run `testZeroValues` on your side to double check? 

I manually generated images to compare, the zero value labels count should have obvious difference, but the UT still pass.

Reply:
tolerance?

Reply:
@potato04 could you try on your side? I'm not sure if it's tolerance issue, but it has obvious diff, I don't think so

Reply:
Same result here,  but i found it's a tolerance issue, the diff of two images calculated by FBSnapshot is 17205/67200≈0.0256(<0.03)

Reply:
below is the comparison code
https://github.com/facebookarchive/ios-snapshot-test-case/blob/3efc829601128f7d9306e8e58db04d32594b62ce/FBSnapshotTestCase/Categories/UIImage%2BCompare.m#L104-L121

Reply:
but the zero value labels is a lot from human eyes.. can't believe it will be very small difference..

Anyway, for zero value tests,  I think we could update the tolerance to make it more strict.